pkg/image/nbdkit.go

```diff
@@ -268,13 +283,20 @@ func waitForNbd(pidfile string) error {
 
 // KillNbdkit stops the nbdkit process
 func (n *Nbdkit) KillNbdkit() error {
+	var err error
 	if n.c == nil {
 		return nil
 	}
 	if n.c.Process != nil {
-		return n.c.Process.Signal(os.Interrupt)
+		err = n.c.Process.Signal(os.Interrupt)
 	}
-	return n.c.Process.Kill()
+	if err != nil {
+		err = n.c.Process.Kill()
```

Comment:
I don't understand this hunk.
Previously, the kill was unconditional, now it relies on signal returning on error, which seems to be a rare scenario that isn't expected to happen.

Reply:
The nbdkit process should be stopped before stopping the log watcher, and the log watcher should be stopped even if the Interrupt or Kill fails. I wanted to otherwise preserve the original behavior of this code by returning the same error it would return without my changes, but it does look confusing.

I think the original code wasn't quite right, either - the Kill would only run if Process was nil, but that would immediately de-reference a nil and crash. If Process was not nil, it would always return whatever Signal returned. I think the original intention was to use Kill as a backup for Signal if it happened to fail for some reason, so I changed it to look more like that.